R/plot_var_compare.R

```diff
@@ -2,82 +2,118 @@
+#' @param obs.shape Shape of observation points. For options see vignette("ggplot2-specs")
+#' @param obs.size Size of observation points. For options see \code{\link[vignette("ggplot2-specs")]{vignette("ggplot2-specs")}}
+#' @param \dots additional arguments passed to \code{ggsave()}
 #'
 #' @seealso Internally uses \link{get_var} and \link{resample_to_field}
 #'
 #'
 #'@examples
-#'sim_folder <- run_example_sim(verbose = FALSE)
-#'nc_file <- file.path(sim_folder, 'output.nc')
-#'nml_file <- file.path(sim_folder, 'glm2.nml')
-#'field_file <- file.path(sim_folder, 'field_data.tsv')
+#'nc_file <- system.file("extdata", "output.nc", package = "glmtools")
+#'field_file <- system.file("extdata", "field_data.tsv", package = "glmtools")
 #'
-#'run_glm(sim_folder)
+#'plot_var_compare(nc_file, field_file, 'temp', resample=FALSE) ##makes a plot
 #'
-#'plot_var_compare(nc_file, field_file, 'temp', resample=FALSE) ##makes a plot!
+#'#Change clor palette, custom legend title, save figure:
+#'plot_var_compare(nc_file, field_file, var_name = 'temp', resample = F, legend.title = 'Temp (*C)',
+#'color.palette = 'PuBuGn', color.direction = 1,fig_path = '~/Downloads/figtest.png', width = 6, height = 8, units = 'in')
 #'
 #'@importFrom akima interp
+#'@importFrom akima interp2xyz
+#'@importFrom gridExtra grid.arrange
+#'@author
+#'Jordan S. Read, Luke A. Winslow, Hilary A. Dugan
 #'@export
-plot_var_compare = function(nc_file, field_file, var_name, fig_path = FALSE, resample = TRUE, col_lim, ...){
+plot_var_compare = function(nc_file, field_file, var_name = 'temp', fig_path = NULL, resample = TRUE, 
+                            legend.title = NULL, interval = 1,method = 'match', text.size = 12,
+                            color.palette = 'RdYlBu', color.direction = -1, 
+                            obs.color = 'white', obs.alpha = 0.6, obs.shape = 16, obs.size = 1,...) {
   
   heatmaps <- .is_heatmap(nc_file, var_name)
   if (!heatmaps){
     warning('plot_var_compare not implemented for 1D variables')
     return()
   }
   
-  start_par = par(no.readonly = TRUE)
-  #Create layout
-  
-  mod_temp = get_var(nc_file, var_name, reference='surface')
-  mod_depths = get.offsets(mod_temp)
-  
+  surface <- get_surface_height(nc_file)
+  max_depth <- max(surface[, 2])
+  min_depth <- 0
+  z_out <- seq(min_depth, max_depth,by = interval) # Set plotting interval
+  mod_temp = get_var(nc_file, var_name, reference='surface',z_out = z_out)
+
+  data = resample_to_field(nc_file, field_file, var_name=var_name, method = method) 
   
-  data = resample_to_field(nc_file, field_file, var_name=var_name, ...)
-  if(resample){
-  	model_df <- resample_sim(mod_temp, t_out = unique(data$DateTime))
-  }else{
+  # Akima interpolation of observed data
+  dataClean = data %>% dplyr::filter_all(all_vars(!is.na(.)))
+
+  df_akima <-interp2xyz(interp(x=as.numeric(dataClean$DateTime), y=dataClean$Depth*1e6, z=dataClean$Observed_temp, duplicate="mean", linear = T,
```

Comment:
`df_akima` is the observed temperature, right? Can the variable be named in a more intuitive way? Also, since it mirrors `model_df`, maybe the two would share naming conventions. Like `observed_df` or something along those lines

Reply:
Good call. `df_akima` was the original object name. But `observed_df` makes a whole lot more sense.